shared/src/commonMain/kotlin/screens/IViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package screens
+
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.receiveAsFlow
+
+/**
+ * Abstract view model which holds the basic functions each view model needs to have.
+ *  [A] represents the UiAction.
+ *  [E] represents the UiEvent.
+ *  [S] represents the UiState which can be a nullable type of Any if no state is applicable.
+ */
+interface IViewModel<A, E, S : Any?> {
```

Comment:
Very good! only I... interfaces are not our convention, try changing this to a ComposeViewModel or BaseComposeViewModel